php-checks/src/main/java/org/sonar/php/checks/CountInsteadOfEmptyCheck.java

```diff
@@ -141,21 +142,17 @@ private boolean containsEmptyCallOnVariable(BinaryExpressionTree logicalExpressi
   // Recursively searches a tree for empty() calls on the target variable
   private boolean containsEmptyCallInTree(Tree tree, ExpressionTree targetVariable) {
     // Check if this node is a function call to empty() with the same variable
-    if (tree.is(Tree.Kind.FUNCTION_CALL)) {
-      FunctionCallTree functionCall = (FunctionCallTree) tree;
-      if (isEmptyFunction(functionCall)) {
-        ExpressionTree emptyArgument = CheckUtils.argument(functionCall, "var", 0)
-          .map(CallArgumentTree::value)
-          .orElse(null);
-        if (emptyArgument != null && areSameVariable(emptyArgument, targetVariable)) {
-          return true;
-        }
+    if (tree.is(Tree.Kind.FUNCTION_CALL) && tree instanceof FunctionCallTree functionCall && isEmptyFunction(functionCall)) {
+      ExpressionTree emptyArgument = CheckUtils.argument(functionCall, "var", 0)
+        .map(CallArgumentTree::value)
+        .orElse(null);
+      if (emptyArgument != null && areSameVariable(emptyArgument, targetVariable)) {
```

Comment:
`orElse` is used here to eliminate the `Optional`. This makes sense in some cases, for example, if we want to call a method that expects the non-optional type. However, we we only want to do a null-check. In this case, I suggest to use the `isPresent` function (see `A value is present only if we have created Optional with a non-null value.` [here](https://www.baeldung.com/java-optional#creation)).
```suggestion
      Optional<ExpressionTree> emptyArgument = CheckUtils.argument(functionCall, "var", 0)
        .map(CallArgumentTree::value);
      if (emptyArgument.isPresent() && areSameVariable(emptyArgument.get(), countArgument)) {
        return true;
```